docs/guides/transactions/running-background-jobs.md

```diff
@@ -45,25 +38,19 @@ namespace StarcounterApplication4
 
 ## Basic Information About Scheduling
 
-Each Starcounter scheduler has a queue of tasks that are supposed to be run on this scheduler. Tasks are picked from the queue and executed. To put a task in a queue, the `Scheduling.RunTask` should be used. When scheduling a task, you can specify the scheduler number, and if the thread should wait for the task to be picked by scheduler and completed. Here is the signature of the `Scheduling.RunTask`:
+Tasks scheduled by `Scheduling.RunTask` as well as ready-to-call handlers make a queue of ready tasks. Then Starcounter picks tasks and runs them on a free thread from the thread pool. By default Starcounter thread pool hosts number of threads equal to number of CPU cores. It allows to maximize throughput by avoiding unnecessary context switches. But what if user code happens to block on IO or synchronization? Starcounter is able to detect such condition and launch an additional thread, so that queue processing is not stall. To maximize throughput it's not recommended to rely on thread pool scaling. Better to avoid blocking calls and use async IO whenever possible. 
+To put a task in a queue, the `Scheduling.RunTask` should be used:
 
 ```csharp
 Task RunTask(Action action)
+Task RunTask(Func<Task> action)
 ```
 
-where `Action action` is a procedure to execute on scheduler.
-
-To make the `Action` execute synchronously, use the `Wait` method:
-
-```csharp
-Scheduling.RunTask(() => { }).Wait()
-```
-
-To determine if current thread is on scheduler call `StarcounterEnvironment.IsOnScheduler()`.
+where `action` is a procedure to execute on a Starcounter thread. Task being returned is a regular .Net task object which can be used in any appropriate context as any regular .Net task.
```

Comment:
> .Net task object which can be used in any appropriate context as any regular .Net task.

What value does "in any appropriate context" add? Why not " .Net task object which can be used as any regular .Net task."?

Reply:
Probably. I just thought that word 'context' includes using task as an operand of await operator.

Reply:
> I just thought that word 'context' includes using task as an operand of await operator.

Okay, then it's probably best to explain that or remove it.